integration-tests/bats/stash.bats

```diff
@@ -352,3 +349,306 @@ teardown() {
+    skip_nbf_ld_1
+    dolt sql -q "CREATE TABLE new_table (id INT PRIMARY KEY);"
+    dolt add .
+    dolt sql -q "INSERT INTO new_table VALUES (1),(2);"
+    dolt sql -q "CREATE TABLE test_table (id INT);"
+    run dolt status
+    [[ "$output" =~ "Changes to be committed:" ]] || false
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ "Untracked files:" ]] || false
+
+    run dolt stash
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Saved working directory and index state" ]] || false
+
+    run dolt status
+    [[ ! "$output" =~ "Changes to be committed:" ]] || false
+    [[ ! "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ "Untracked files:" ]] || false
+
+    run dolt ls
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "test_table" ]] || false
+
+    # popping the stash should result in no modified working set, but with staged new_table
+    run dolt stash pop
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Changes to be committed:" ]] || false
+    [[ ! "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ "$output" =~ "Untracked files:" ]] || false
+
+    run dolt ls
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "new_table" ]] || false
+    [[ "$output" =~ "test_table" ]] || false
+}
+
+@test "stash: stashing working set with deleted table and popping it" {
+    skip_nbf_ld_1
+    dolt sql -q "CREATE TABLE new_table (id INT PRIMARY KEY);"
+    dolt commit -Am "create new table"
+
+    dolt sql -q "DROP TABLE new_table;"
+    run dolt status
+    [[ ! "$output" =~ "Changes to be committed:" ]] || false
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ ! "$output" =~ "Untracked files:" ]] || false
+
+    run dolt stash
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Saved working directory and index state" ]] || false
+
+    run dolt status
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+    run dolt ls
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "new_table" ]] || false
+
+    # popping the stash should result in no modified working set, but with staged new_table
+    run dolt stash pop
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ "Changes to be committed:" ]] || false
+    [[ "$output" =~ "Changes not staged for commit:" ]] || false
+    [[ ! "$output" =~ "Untracked files:" ]] || false
+
+    run dolt ls
+    [[ ! "$output" =~ "new_table" ]] || false
+}
+
+@test "stash: simple stashing and popping stash after running GC" {
+    skip_nbf_ld_1
+    dolt sql -q "INSERT INTO test VALUES (1, 'a')"
+    run dolt sql -q "SELECT * FROM test"
+    [ "$status" -eq 0 ]
+    result=$output
+
+    run dolt stash
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Saved working directory and index state" ]] || false
+
+    run dolt sql -q "SELECT * FROM test"
+    [ "$status" -eq 0 ]
+    [ "$output" = "" ]
+
+    dolt gc
```

Comment:
oh! very nice! was GC already smart enough to follow all the datasets in the database and see that the stashed rootvalues are still in use? If you create a dolt commit after the stash, does that change anything? (without knowing anything about our gc internals... I'm just wondering if we're getting lucky with that rootvalue being the most recent one created or something)